src/Tools/ExternalAccess/Razor/Cohost/RazorDynamicRegistrationServiceFactory.cs

```diff
@@ -45,7 +45,10 @@ public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestCon
             }
 
             // We use a string to pass capabilities to/from Razor to avoid version issues with the Protocol DLL
-            var serializedClientCapabilities = JsonConvert.SerializeObject(clientCapabilities);
+
+            var serializedClientCapabilities = JsonConvert.SerializeObject(clientCapabilities is VSInternalClientCapabilities internalClientCapabilities
+                ? internalClientCapabilities
+                : clientCapabilities);
```

Comment:
I'm not sure if this matters - I *think* even if its a VSInternalClientCapabilities it will get serialized that way, even if its passed in as a ClientCapabilities. 
 I'm not 100% sure of that statement though

What I originally meant was wondering if you need to ensure the VS custom converters are added to your jsonserializer instance - https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core/LanguageServer/AbstractInProcLanguageClient.cs#L204

Reply:
Oh right, derp, of course. Yes, looks like that would be needed for text document registration options, if nothing else